$OpenBSD: patch-libavcodec_Makefile,v 1.9 2010/05/26 21:29:56 jakemsr Exp $

Anything requiring cabac.h will only build with -O -fomit-frame-pointer.

--- libavcodec/Makefile.orig	Mon Mar  8 04:33:02 2010
+++ libavcodec/Makefile	Mon Mar 22 15:20:40 2010
@@ -729,6 +729,45 @@ CLEANFILES = sin_tables.c cos_tables.c *_tables.h *_ta
 
 include $(SUBDIR)../subdir.mak
 
+ifdef ARCH_X86_32
+ifeq (,$(findstring -O,$(CFLAGS)))
+x86/dsputil_mmx.o: CFLAGS += -O
+x86/flacdsp_mmx.o: CFLAGS += -O
+x86/fft_3dn.o: CFLAGS += -O
+x86/fft_3dn2.o: CFLAGS += -O
+x86/mlpdsp.o: CFLAGS += -O
+x86/snowdsp_mmx.o: CFLAGS += -O
+h264.o: CFLAGS += -O
+h264_parser.o: CFLAGS += -O
+h264_loopfilter.o: CFLAGS += -O
+h264_direct.o: CFLAGS += -O
+h264_sei.o: CFLAGS += -O
+h264_ps.o: CFLAGS += -O
+h264_refs.o: CFLAGS += -O
+h264_cavlc.o: CFLAGS += -O
+h264_cabac.o: CFLAGS += -O
+svq3.o: CFLAGS += -O
+cabac.o: CFLAGS += -O
+endif
+x86/dsputil_mmx.o: CFLAGS += -fomit-frame-pointer
+x86/flacdsp_mmx.o: CFLAGS += -fomit-frame-pointer
+x86/fft_3dn.o: CFLAGS += -fomit-frame-pointer
+x86/fft_3dn2.o: CFLAGS += -fomit-frame-pointer
+x86/mlpdsp.o: CFLAGS += -fomit-frame-pointer
+x86/snowdsp_mmx.o: CFLAGS += -fomit-frame-pointer
+h264.o: CFLAGS += -fomit-frame-pointer
+h264_parser.o: CFLAGS += -fomit-frame-pointer
+h264_loopfilter.o: CFLAGS += -fomit-frame-pointer
+h264_direct.o: CFLAGS += -fomit-frame-pointer
+h264_sei.o: CFLAGS += -fomit-frame-pointer
+h264_ps.o: CFLAGS += -fomit-frame-pointer
+h264_refs.o: CFLAGS += -fomit-frame-pointer
+h264_cavlc.o: CFLAGS += -fomit-frame-pointer
+h264_cabac.o: CFLAGS += -fomit-frame-pointer
+svq3.o: CFLAGS += -fomit-frame-pointer
+cabac.o: CFLAGS += -fomit-frame-pointer
+endif
+
 $(SUBDIR)dct-test$(EXESUF): $(SUBDIR)dctref.o
 
 $(SUBDIR)cos_tables.c: $(SUBDIR)costablegen$(HOSTEXESUF)
